### PR TITLE
Fix aliasing function type returning a Type with TypeSuffixes

### PR DESCRIPTION
**Pull request by ntrel**
This merges function type alias and type alias parsing, leading to simpler code.
A function type with a ~~_TypeCtor_ (or other~~ invalid storage class ~~)~~ is now deprecated, ~~just like a function pointer type with a _TypeCtor_.~~

Because the merge affected two tests' errors a bit:
* I made the check to parse `auto ref` starting a function literal less strict, causing a better error message when `ref` is missing.
* I preserved the error when detecting an identifier after *Type*.

### CODE REVIEW

File: compiler/test/compilable/issue16020.d [hunk 1]
```diff
@@ -19,21 +20,13 @@ alias void F51() @system;
...
-alias F63 = const shared int();
-static assert (is(F61 == F62));
-static assert (is(F63 == F62));
-
-alias F71 = int() immutable inout;
-alias int F72() immutable inout;
-alias F73 = immutable inout int();
-static assert (is(F71 == F72));
-static assert (is(F73 == F72));
-
```
Comment (thewilsonator) [inline compiler/test/compilable/issue16020.d:33]: stupid question: why are these removed?

Comment (ntrel) [reply compiler/test/compilable/issue16020.d]: I've reverted this now. I forgot that a method type can have TypeCtor attributes.

**Review by thewilsonator — APPROVED** [review]
